Apps/AdminWebClient/src/ClientApp/src/components/core/EmailCommunication.vue

```diff
@@ -0,0 +1,176 @@
+                            v-model="content"
+                            :toolbar-attributes="{ color: 'gray' }"
+                            placeholder="Write the email content here..."
+                            :extensions="extensions"
+                        />
+                    </v-col>
+                </v-row>
+            </v-form>
+        </v-card-text>
+        <!-- Buttons -->
+        <v-card-actions>
+            <v-spacer></v-spacer>
+            <v-btn color="blue darken-1" text @click="send()">Send</v-btn>
+        </v-card-actions>
+    </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch, Emit } from "vue-property-decorator";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
+import BannerFeedback from "@/models/bannerFeedback";
+import {
+    TiptapVuetify,
+    Heading,
+    Bold,
+    Italic,
+    Strike,
+    Underline,
+    Code,
+    Paragraph,
+    BulletList,
+    OrderedList,
+    ListItem,
+    Link,
+    Blockquote,
+    HardBreak,
+    History
+} from "tiptap-vuetify";
+import { ResultType } from "@/constants/resulttype";
+import { ICommunicationService } from "@/services/interfaces";
+
+@Component({
+    components: {
+        TiptapVuetify
+    }
+})
+export default class EmailCommunication extends Vue {
+    private isLoading: boolean = false;
+    private showFeedback: boolean = false;
+    private bannerFeedback: BannerFeedback = {
+        type: ResultType.NONE,
+        title: "",
+        message: ""
+    };
+    private priorityItems = ["Urgent", "Medium", "Low"];
+
+    private subject: string = "";
+    private priority: string = "";
+    private extensions: any = [
+        History,
+        Blockquote,
+        Link,
+        Underline,
+        Strike,
+        Bold,
+        Italic,
+        ListItem,
+        BulletList,
+        OrderedList,
+        [
+            Heading,
+            {
+                options: {
+                    levels: [1, 2, 3, 4]
+                }
+            }
+        ],
+        Bold,
+        Code,
+        Paragraph,
+        HardBreak
+    ];
+    private content: string = "<p></p>";
+
+    private contentValid(): boolean {
+        return this.content.replace("<[^>]*>", "") !== "" ? true : false;
+    }
+
+    private send() {
+        this.isLoading = true;
+        this.isFinishedLoading();
+        if (
+            (this.$refs.form as Vue & { validate: () => boolean }).validate() &&
+            this.contentValid()
+        ) {
+            // NETWORK REQUEST GOES HERE!
+            // Temp data below.
+            this.showFeedback = true;
+            this.bannerFeedback = {
+                type: ResultType.Success,
+                title: "Success",
+                message: "Email sent."
+            };
+            this.subject = "";
+            this.priority = "";
+            this.content = "<p></p>";
+            (this.$refs.form as Vue & {
```

Comment:
Is there a type that has said method? if so lets cast it to that instead

Reply:
Not from my research